src/uct/ib/base/ib_iface.c

```diff
@@ -695,10 +743,17 @@ static ucs_status_t uct_ib_iface_init_pkey(uct_ib_iface_t *iface,
 
     if (!pkey_found) {
         if (config->pkey_value == UCS_HEXUNITS_AUTO) {
-            ucs_error("There is no valid pkey with full membership on "
-                      UCT_IB_IFACE_FMT, UCT_IB_IFACE_ARG(iface));
+            if (!allow_limited && has_pkey_limited) {
+                ucs_debug("there is no valid pkey with full membership on "
+                          UCT_IB_IFACE_FMT, UCT_IB_IFACE_ARG(iface));
+                /* retry the search allowing PKEY with limited membership */
+                allow_limited = 1;
+                goto retry;
```

Comment:
maybe simply and use helper function?
e.g 
1. search_for_pkey(full)
2. search_for_pkey(limited)